src/chat/tools/listTemplates.ts

```diff
@@ -43,6 +43,14 @@ export class ListTemplatesTool extends BaseLanguageModelTool<IListTemplatesParam
       const templateSummary = summarizeProjectTemplate(template);
       templateStrings.push(new LanguageModelTextPart(templateSummary));
     });
+    if (inputTagsPassed && !templateStrings.length) {
+      // invalid tags, no results
+      return new LanguageModelToolResult([
+        new LanguageModelTextPart(
+          `No templates matched the provided tags: ${JSON.stringify(params.tags)}`,
+        ),
+      ]);
+    }
```

Comment:
Unrelated to this PR, but noticed when the model attempted to pass hallucinated/invalid tags to filter templates and no result was returned (causing an error 400 response)